packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -161,26 +154,31 @@ function routerReducer(
 			screens = removeScreen( state, action.screen );
 			break;
 		case 'goto':
-			locationHistory = goTo( state, action.path, action.options );
+			const goToNewState = goTo( state, action.path, action.options );
+			currentLocation = goToNewState.currentLocation;
+			focusSelectors = goToNewState.focusSelectors;
 			break;
 		case 'gotoparent':
-			locationHistory = goToParent( state, action.options );
+			const goToParentNewState = goToParent( state, action.options );
+			currentLocation = goToParentNewState.currentLocation;
+			focusSelectors = goToParentNewState.focusSelectors;
 			break;
 	}
 
+	if ( currentLocation?.path === state.initialPath ) {
+		currentLocation = { ...currentLocation, isInitial: true };
+	}
```

Comment:
Do you really want to do this? `isInitial` should be `true` only when the page is initially loaded. Then we don't want to do animations or change focus. But when later navigating to the initial page, the initial page is a page like any other, we want to run animations and to restore focus.

Reply:
> Do you really want to do this?

No, that's the wrong logic. I had already noticed it while working on the revamped animation, and already prepared a fix (https://github.com/WordPress/gutenberg/pull/64777/commits/0fbe6a345fff180344c1d3e069e1fd20289cf990) which I'm going to extract to a separate PR